### PR TITLE
feat: [OI-249] login data status

### PR DESCRIPTION
**Pull request by mmoio**
- separate api functions
- separate login data hook
- separate login buttons
- adds react-query
- refactors api
- refactors data fetching
- adds spid skeletons
- reduces code repeatition
  SpidSelection & SpidModal
- login refactoring
- improves error handling
- udated unit tests

image

image

### CODE REVIEW

**Comment by sebbalex** [comment]
I am not particularly convinced about loading the spinner when the page loads. This causes the user to see the loading every time they land on the page when in fact the time it takes the user to click on the “Entra con SPID” button can be used for background loading.
I would move the spinner inside the modal (and dialog) so that this does not happen.
wdyt?

**Comment by sebbalex** [comment]
as a secondary issue, when a client_id is not present in GET I see in console this message, maybe we should think about something more clear

image

**Comment by mmoio** [comment]
> as a secondary issue, when a client_id is not present in GET I see in console this message, maybe we should think about something more clear
> 
> image

I'have changed the error logic, using react-query, but errors should still be handled in some way within the ui/ux. I think it deserves a separate PR

**Review by sebbalex — APPROVED** [review]
:rocket:

File: src/oneid/oneid-ecs-core/src/main/webui/src/pages/login/index.tsx [hunk 1]
```diff
@@ -219,29 +127,29 @@ const Login = () => {
...
                 style={{
                   width: '100%',
                   maxWidth: '100px',
                   maxHeight: '100px',
                   objectFit: 'cover',
                 }}
-                src={clientData?.logoUri}
-                alt={clientData?.friendlyName}
+                src={clientQuery.data?.logoUri}
+                alt={clientQuery.data?.friendlyName || 'PagoPa Logo'}
```
Comment (sebbalex) [inline src/oneid/oneid-ecs-core/src/main/webui/src/pages/login/index.tsx:147]:
```suggestion
                alt={clientQuery.data?.friendlyName || 'PagoPA Logo'}
```

**Review by sebbalex — APPROVED** [review]

**Pull request merged**